packages/client/src/runtime/query.ts

```diff
@@ -767,7 +765,9 @@ ${indent(value.toString(), 2)}
   }
 }
 
-export type ArgValue = string | boolean | number | undefined | Args | string[] | boolean[] | number[] | Args[] | null
+export type ArgValueScalar = string | boolean | number | Args
+
+export type ArgValue = ArgValueScalar | ArgValueScalar[] | undefined | null
```

Comment:
It's not obvious what this influences from the diff as the new type isn't used explicitly, but it fixed a compiler warning on line 684:

```ts
681    this.hasError =
682      Boolean(error) ||
683      (value instanceof Args ? value.hasInvalidArg : false) ||
684 ->   (Array.isArray(value) && value.some((v) => (v instanceof Args ? v.hasInvalidArg : false)))
                                              ^
                                         here |
```

TypeScript was able to infer the correct type of `value` after the `isArray` check, but it gave up on `v`.

Initially I had extracted `ArgValueScalar` to annotate `v` explicitly with it, but just extracting it and thus simplifying `ArgValue` turned out to be already enough for TypeScript to infer the correct type here.